[CI] SpecificMasterNodesIT#testElectOnlyBetweenMasterNodes fails

Seen at least three times in master in the last few days:

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+internalClusterTest/497/console
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/1777/console
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/1710/console

Latest reproduce line:
```
./gradlew :server:integTest \
  -Dtests.seed=2CFBF83AF54213EC \
  -Dtests.class=org.elasticsearch.cluster.SpecificMasterNodesIT \
  -Dtests.method="testElectOnlyBetweenMasterNodes" \
  -Dtests.security.manager=true \
  -Dtests.locale=ar-DZ \
  -Dtests.timezone=Africa/Kinshasa \
  -Dcompiler.java=11 \
  -Druntime.java=8
```

Running repeatedly on Ubuntu didn't reproduce after 50 runs.

Logs contain NPEs:
```
14:52:15   1> java.lang.NullPointerException: null
14:52:15   1> 	at org.elasticsearch.test.InternalTestCluster.getMasterName(InternalTestCluster.java:1913) ~[framework-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
14:52:15   1> 	at org.elasticsearch.test.InternalTestCluster.getMasterName(InternalTestCluster.java:1902) ~[framework-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
14:52:15   1> 	at org.elasticsearch.test.InternalTestCluster.nonMasterClient(InternalTestCluster.java:776) ~[framework-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
14:52:15   1> 	at org.elasticsearch.cluster.SpecificMasterNodesIT.lambda$testElectOnlyBetweenMasterNodes$0(SpecificMasterNodesIT.java:122) ~[test/:?]
14:52:15   1> 	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:846) [framework-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
```
But there are also several earlier connection Exceptions like:

```
14:52:14   1> Caused by: java.net.ConnectException: Connection refused
```

--- a/server/src/test/java/org/elasticsearch/cluster/SpecificMasterNodesIT.java
+++ b/server/src/test/java/org/elasticsearch/cluster/SpecificMasterNodesIT.java
@@ -84,7 +84,6 @@ public class SpecificMasterNodesIT extends ESIntegTestCase {
             .execute().actionGet().getState().nodes().getMasterNode().getName(), equalTo(nextMasterEligibleNodeName));
     }
 
-    @AwaitsFix(bugUrl="https://github.com/elastic/elasticsearch/issues/38331")
     public void testElectOnlyBetweenMasterNodes() throws Exception {
         internalCluster().setBootstrapMasterNodeIndex(0);
         logger.info("--> start data node / non master node");

--- a/test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java
@@ -167,6 +167,7 @@ import static org.hamcrest.Matchers.greaterThanOrEqualTo;
 import static org.hamcrest.Matchers.not;
 import static org.hamcrest.Matchers.nullValue;
 import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertThat;
 import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
@@ -1909,8 +1910,9 @@ public final class InternalTestCluster extends TestCluster {
     public String getMasterName(@Nullable String viaNode) {
         try {
             Client client = viaNode != null ? client(viaNode) : client();
-            ClusterState state = client.admin().cluster().prepareState().execute().actionGet().getState();
-            return state.nodes().getMasterNode().getName();
+            final DiscoveryNode masterNode = client.admin().cluster().prepareState().get().getState().nodes().getMasterNode();
+            assertNotNull(masterNode);
+            return masterNode.getName();
         } catch (Exception e) {
             logger.warn("Can't fetch cluster state", e);
             throw new RuntimeException("Can't get master node " + e.getMessage(), e);

